cmd/juju/machine/remove.go

```diff
@@ -133,12 +156,16 @@ func (c *removeCommand) Run(ctx *cmd.Context) error {
 	}
 	defer client.Close()
 
-	destroy := client.DestroyMachines
-	if c.Force {
-		destroy = client.ForceDestroyMachines
+	var results []params.DestroyMachineResult
+	if c.KeepInstance {
```

Comment:
The issue is that there are 2 facades that could be used - api/client and api/machinemanager. Only quite recent versions of Juju have the machinemanager api. So it's easier to use the knowledge that if keep is specified and we have the relevant machinemanager facade version available, call that directly. Otherwise we need to fall back to using the removeMachineAdapter which knows how to translate the differing return results from the api/client and api/machinemanager implementations.